pkg/controller/volume/persistentvolume/pv_controller.go

```diff
@@ -361,17 +361,42 @@ func (ctrl *PersistentVolumeController) syncUnboundClaim(ctx context.Context, cl
 				return nil
 			}
 
+			// The claim can't be bound to any existing volume. let's check if it can be
+			// dynamically provisioned. If can't, we will emit an event to inform the user
+			// the reason why the claim is pending. otherwise, we will provision a volume
+			// for the claim in an asynchronous goroutine and return immediately.
 			switch {
 			case delayBinding && !storagehelpers.IsDelayBindingProvisioning(claim):
+				// If the claim is in delay binding mode and not annotated with
+				// "volume.kubernetes.io/selected-node" by the scheduler's VolumeBinding
+				// plugin, it means that no active pod is observed by the scheduler to
+				// be using this claim. In this case, we should emit an event to inform
+				// the user that the claim is waiting for a pod to be scheduled.
```

Comment:
The whole "_If the claim is in delay binding mode and not annotated with"volume.kubernetes.io/selected-node" by the scheduler's VolumeBinding plugin_" just repeats the `case` condition above. There is no need to repeat that. Describe what it means, i.e. just "_No active pod is observed by the scheduler to be using this claim ..._".

* Active case could be better ('_Scheduler does not observe any pod using this claim_").

* Don't describe what should happen, describe what happens - "_Emit an event ..,_" And I would argue, is function name `emitEventForUnboundDelayBindingClaim` not clear enough?